applications/sandbox/systems/pointcloud_particlesystem.hpp

```diff
@@ -0,0 +1,69 @@
+#pragma once
+#include <rendering/data/particle_system_base.hpp>
+
+using namespace legion;
+
+struct pointCloudParameters
+{
+    math::vec3 startingSize;
+    rendering::material_handle particleMaterial;
+    rendering::model_handle particleModel;
+
+    bool looping = false;
+    float maxLifeTime = 0.0f;
+    float startingLifeTime = 0.0f;
+    uint spawnRate = 0;
+    uint particleCount = 0;
+    uint maxParticles = 0;
+    math::vec3 startingVel = math::vec3(0);
+    float sizeOverLifeTime = 0.0f;
+};
+
+class PointCloudParticleSystem : public rendering::ParticleSystemBase
+{
+public:
+    PointCloudParticleSystem(pointCloudParameters params, const std::vector<math::vec3>& positions)
+    {
+        m_looping = params.looping;
+        m_maxLifeTime = params.maxLifeTime;
+        m_startingLifeTime = params.startingLifeTime;
+        m_spawnRate = params.spawnRate;
+        m_particleCount = params.particleCount;
+        m_maxParticles = params.maxParticles;
+        m_startingVelocity = params.startingVel;
+        m_startingSize = params.startingSize;
+        m_sizeOverLifetime = params.sizeOverLifeTime;
+        m_particleMaterial = params.particleMaterial;
+        m_particleModel = params.particleModel;
+        m_positions = positions;
+    }
```

Comment:
in C++ we can do 
```cpp
PointCloudParticleSystem(pointCloudParameters params, const std::vector<math::vec3>& positions) : 
    m_looping(params.loop), m_maxLifeTime(params.maxLifeTime)
{
    // Your normal code here
}
```
It can make it a little more readable